### ISSUE TITLE
Support browser extensions in create flow

### ISSUE THREAD

**Issue by GBKS** [GBKS/nosta-me#16]
The create flow can only be used with a newly created key. Using an external signer is generally better and should be supported.

Problem is just that external signers right now usually have a crappy or non-existent key generation user flow. This is what Nosta is trying to fix.

Will have to think of something here...

**Comment by neilck** [comment]
Did you need a key generation user flow like on nosta.me but in an extension?

I'm currently working on a browser extension that works like nos2x but supports multiple keys.
https://github.com/neilck/aka-extension.

Ideally a user could generate a private key in my extension, I could then redirect them to nosta.me to finish creating their profile, and then display their profile in my extension (instead of just the keys values).

**Comment by GBKS** [comment]
Yes, that is something I need to look into. A user should be able to arrive on Nosta with keys, but otherwise no data on Nostr, and then enter the profile setup flow from there.

Parts of this already exist - you can log in and go to your empty profile. That empty state should give you a few options:
- Set up your profile from scratch via the onboarding flow
- Search more relays (maybe you have profile data somewhere, the site just didn't find it)

And a deep-link option, for apps like yours also sounds like a good idea. I haven't thought through this very deeply and don't have designs yet, but sounds like something I should get to sooner than later.

Question for you. If a user changes their "active key" in your extension, do you have a way to inform websites about it?

**Comment by neilck** [comment]
> Yes, that is something I need to look into. A user should be able to arrive on Nosta with keys, but otherwise no data on Nostr, and then enter the profile setup flow from there.
> 
> Parts of this already exist - you can log in and go to your empty profile. That empty state should give you a few options:
> 
> * Set up your profile from scratch via the onboarding flow
> * Search more relays (maybe you have profile data somewhere, the site just didn't find it)
The hand-off to nostra.me after a new private key is generated I think is easy, just redirect to nosta.me/<pubkey>. Nosta.me guides new user to create new profile, and calls extension to sign the metadata event. The extension then stored the metadata locally for display in the extension.
> 
> Question for you. If a user changes their "active key" in your extension, do you have a way to inform websites about it?
It's currently not specified in NIP-07. An app could manually check by requesting the public key again. 
I think it would be better for the extension to send a public key changed DOM event. 
I know it was suggested somewhere else there should also be a DOM event to inform an app that the window.nostr script element has been added and is ready for calling.
